src/app/Caches/Create/Features/SecuredCache.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState, useEffect } from 'react';
+import {
+    Divider,
+    Form,
+    Text,
+    TextContent,
+    TextVariants,
+    Select,
+    SelectOption,
+    SelectVariant,
+    FormGroup,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { MoreInfoTooltip } from '@app/Common/MoreInfoTooltip';
+import { ConsoleServices } from '@services/ConsoleServices';
+
+const SecuredCache = (props: {
+    securedOptions: SecuredCache,
+    securedOptionsModifier: (SecuredCache) => void,
+}) => {
+
+    const { t } = useTranslation();
+    const brandname = t('brandname.brandname');
+
+    //Secured Cache
+    const [roles, setRoles] = useState<string[]>(props.securedOptions.roles);
+
+    const [loading, setLoading] = useState(true);
+    const [availableRoles, setAvailableRoles] = useState<string[]>([]);
+    const [error, setError] = useState<'success' | 'error' | 'default'>('default');
+
+    const [isOpenRoles, setIsOpenRoles] = useState(false);
+
+    useEffect(() => {
+        if (loading) {
+            ConsoleServices.caches()
+                .getSecurityRoles()
+                .then((r) => {
+                    if (r.isRight()) {
+                        setAvailableRoles(r.value);
+                        setError('success')
+                    }
+                    else {
+                        setError('error');
+                    }
+                }).then(() => setLoading(false));
+        }
+    }, [loading]);
+
+    useEffect(() => {
+        props.securedOptionsModifier({
+            roles: roles,
+        });
+    }, [roles]);
+
+    // Options for Isolation Level
+    const rolesOptions = () => {
+        const options = availableRoles.map((role) => (
+            <SelectOption key={role} value={role} />
+        ))
+        return options
+    };
+
+    const onSelectRoles = (event, selection) => {
+        if (roles.includes(selection))
+            setRoles(roles.filter(role => role !== selection));
+        else
+            setRoles([...roles, selection]);
+    }
+
+    return (
+        <Form>
```

Comment:
check the Card and CardBody changes I reported in indexing to change this accordingly.